### ISSUE TITLE
Web worker on CORS workaround

### ISSUE THREAD

**Issue by ghetolay** [mozilla/pdf.js#7304]
Hi,

I have some project using web worker and I also came across the CORS problem. I solved it using ajax :  
I download the worker file using ajax then create a blob URL to refer to the downloaded worker. then you can use that blob URL like any URL to create web workers.
So to work with CORS you just need to setup ajax CORS.

The only downside is that you can't create web workers until the ajax call is done. Here is some code :
```javascript
var promise = new Promise( function(resolve, reject){
    var ajax = new XMLHttpRequest();
    ajax.open("GET", 'http://url_to_web_worker_involving_CORS', true);
    ajax.onreadystatechange = function (oEvent) {  
        if (ajax.readyState === 4) {  
            if (ajax.status === 200) 
               resolve(
                    URL.createObjectURL(
		        new Blob([ajax.responseText], {type: 'application/javascript'}));
               );
            else
              reject( ajax.statusText );  
       }  
    }; 
    ajax.send();

//worker can only be created once our ajax call has ended
promise.then( function(workerUrl){
    worker = new worker( workerUrl );
};
```

Also we may check if CORS is involved before doing so, it's something I planned to do but I don't really know how to do it. 

PDF.js is a rather large project and I've never look on the code so I didn't try to create a PR. I'm pretty sure you'll be more efficient at detecting what must be adapt and how to do it correctly.

**Comment by yurydelendik** [comment]
Not enough information is provided (such as version of PDF.js you are using, browser, etc), but looks like you want fix something #6753 fixed. Closing as invalid.

**Comment by ghetolay** [comment]
Ho I missed this last pre-release, will try it.
